src/components/data-details/index.js

```diff
@@ -1,48 +1,121 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import ItemAttribute from '../item-attribute'
+import FormAttribute from '../form-attribute'
 
 import './styles.scss'
 
-const DataDetails = ({
-  children,
-  formattedData,
-  type,
-}) => (
-  <div className="data-details">
-    {formattedData.map((row, idx) => (
-      <div className="row" key={idx}>
-        {row.map(item => (
-          <ItemAttribute
-            attribute={item.attribute}
-            key={item.attribute}
-            value={item.value}
-            type={type}
-          />
-        ))}
+export default class DataDetails extends React.Component {
+  constructor() {
+    super()
+    this.state = {
+      selectedEntity: null,
+    }
+  }
+
+  componentWillMount() {
+    this.setState({selectedEntity: this.props.currentEntity})
+  }
+
+  handleUpdate(attribute, value) {
+    this.setState({selectedEntity: {
+      ...this.state.selectedEntity,
+      [attribute]: value.trim(),
+    }})
+  }
+
+  handleCancelUpdate(toggleEdit) {
+    this.setState({selectedEntity: this.props.currentEntity})
+    toggleEdit()
+  }
+
+  render() {
+    const {
+      // children,
+      currentEntity,
+      editing,
+      formattedData,
+      toggleEdit,
+      type,
+      updateEntity,
+    } = this.props
+    return (
+      <div className="data-details">
+        <div className="info">
+          <form
+            onSubmit={e => {
+              console.log('On Submit')
+              e.preventDefault()
+              if(editing)
+                return
+              console.log('Submitting')
+              updateEntity(currentEntity.id, this.state.selectedEntity)
```

Comment:
Déclare les variabkes de ton state comme pour les `this.props` au début de ta fonction